src/Symfony/Component/HtmlSanitizer/TextSanitizer/UrlSanitizer.php

```diff
@@ -132,7 +132,7 @@ private static function matchAllowedHostParts(array $uriParts, array $trustedPar
     {
         // Check each chunk of the domain is valid
         foreach ($trustedParts as $key => $trustedPart) {
-            if ($uriParts[$key] !== $trustedPart) {
+            if (array_key_exists($key, $uriParts) && $uriParts[$key] !== $trustedPart) {
```

Comment:
shouldn't this fail if they don't match?
```suggestion
            if (!array_key_exists($key, $uriParts) || $uriParts[$key] !== $trustedPart) {
```

Reply:
That's indeed better.